k8s-api/src/main/java/io/enmasse/k8s/api/ConfigMapAddressApi.java

```diff
@@ -64,9 +66,10 @@ private Address getAddressFromConfig(ConfigMap configMap) {
                 builder.setUid(configMap.getMetadata().getUid());
             }
 
-            if (address.getResourceVersion() == null) {
-                builder.setResourceVersion(configMap.getMetadata().getResourceVersion());
-            }
+// resourceVersion is server assigned when the update is received, so any value from the config.json will always be stale.
```

Comment:
I think that with this change, address setResourceVersion is never called, so the config.json will never contain that info, which means lockResourceVersion(address.getResourceVersion()) will always be called with 'null' as the argument.

How it has worked previously is that address resourceVersion always gets set on deserialization  like above, so it will always match the resourceVersion of the underlying configmap. So i would instead remove the 'if address.getResourceVersion() == null), because it should always be set to the configmap resourceversion IMO.

Reply:
You are indeed right and it was the change that I had been testing.   I reverted too much when adding the comments just before making the PR.